packages/components/src/components/segmented-control/segmented-control-item/segmented-control-item-styles.ts

```diff
@@ -23,25 +27,34 @@ export const getColors = (
   isSelected: boolean,
   bgColor: SegmentedControlBackgroundColor,
   theme: Theme
-): { backgroundColor: string; buttonColor: string; labelColor: string } => {
+): {
+  backgroundColor: string;
+  buttonColor: string;
+  labelColor: string;
+  borderColor: string;
+  hoverBorderColor: string;
+} => {
   const themedColors = getThemedColors(theme);
-  const { primaryColor, contrastMediumColor } = isSelected ? getInvertedThemedColors(theme) : themedColors;
 
-  const backgroundColor =
-    themedColors[
-      isSelected ? 'contrastHighColor' : bgColor === 'background-surface' ? 'backgroundColor' : 'backgroundSurfaceColor'
-    ];
+  const { primaryColor, contrastMediumColor } = themedColors;
+  const backgroundColor = themedColors[bgColor === 'background-surface' ? 'backgroundColor' : 'backgroundSurfaceColor'];
+  const borderColor = isSelected ? themedColors.primaryColor : isDisabled ? themedColors.disabledColor : 'transparent';
```

Comment:
Could use destructured `primaryColor` and destructure `disabledColor`, too.

Reply:
Fixed it